test/e2e/install_test.go

```diff
@@ -43,158 +43,182 @@ var _ = Describe("Operator Install", func() {
 		})
 		AfterEach(func() {
 			err := c.Delete(ctx, operator)
 			Expect(err).ToNot(HaveOccurred())
-			Eventually(func(g Gomega) {
-				err = c.Get(ctx, types.NamespacedName{Name: operatorName}, &operatorv1alpha1.Operator{})
-				Expect(errors.IsNotFound(err)).To(BeTrue())
-			}).WithTimeout(defaultTimeout).WithPolling(defaultPoll).Should(Succeed())
 
-			err = c.Delete(ctx, operatorCatalog)
+			err = c.Delete(ctx, &catalogd.Catalog{ObjectMeta: metav1.ObjectMeta{Name: registryV1CatalogName}})
 			Expect(err).ToNot(HaveOccurred())
-			Eventually(func(g Gomega) {
-				err = c.Get(ctx, types.NamespacedName{Name: operatorCatalog.Name}, &catalogd.Catalog{})
-				Expect(errors.IsNotFound(err)).To(BeTrue())
-			}).WithTimeout(defaultTimeout).WithPolling(defaultPoll).Should(Succeed())
-
-			// speed up delete without waiting for gc
-			err = c.DeleteAllOf(ctx, &catalogd.BundleMetadata{})
-			Expect(err).ToNot(HaveOccurred())
-			err = c.DeleteAllOf(ctx, &catalogd.Package{})
-			Expect(err).ToNot(HaveOccurred())
-
-			Eventually(func(g Gomega) {
-				// ensure resource cleanup
-				packages := &catalogd.PackageList{}
-				err = c.List(ctx, packages)
-				Expect(err).To(BeNil())
-				Expect(packages.Items).To(BeEmpty())
-
-				bmd := &catalogd.BundleMetadataList{}
-				err = c.List(ctx, bmd)
-				Expect(err).To(BeNil())
-				Expect(bmd.Items).To(BeEmpty())
-
-				err = c.Get(ctx, types.NamespacedName{Name: operatorName}, &rukpakv1alpha1.BundleDeployment{})
-				Expect(errors.IsNotFound(err)).To(BeTrue())
-			}).WithTimeout(5 * time.Minute).WithPolling(defaultPoll).Should(Succeed())
+		})
+		It("installs the specified package with correct bundle path", func() {
+			By("eventually reporting a successful resolution")
+			Eventually(checkOperatorResolved(ctx, operator)).WithTimeout(defaultTimeout).WithPolling(defaultPoll).Should(Succeed())
+
+			By("eventually reporting a successful installation")
+			Eventually(checkOperatorInstalled(ctx, operator)).WithTimeout(defaultTimeout).WithPolling(defaultPoll).Should(Succeed())
+
+			By("verifying the expected bundle path")
+			bd := rukpakv1alpha1.BundleDeployment{}
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(operator), &bd)).To(Succeed())
+			Expect(bd.Spec.Template.Spec.Source.Type).To(Equal(rukpakv1alpha1.SourceTypeImage))
+			Expect(bd.Spec.Template.Spec.Source.Image.Ref).To(Equal("localhost/testdata/bundles/registry-v1/prometheus-operator:v0.47.0"))
+			Expect(operator.Status.ResolvedBundleResource).To(Equal(bd.Spec.Template.Spec.Source.Image.Ref))
+			Expect(operator.Status.InstalledBundleResource).To(Equal(bd.Spec.Template.Spec.Source.Image.Ref))
 		})
 	})
+	// TODO: Turn this test on once a operators in the plain bundle format are installable from a catalog, and use the
+	// plain bundle included in a separate catalog than the registryV1-catalog
+
+	// When("An operator installation is attempted that isn't available in any catalog", func() {
```

Comment:
Should we remove these comments from this PR and have them added back uncommented in the plain-bundle-support PR? Though it seems these are the tests Ankita added, but now they need plain-bundle support? ... Maybe it's because it's late on Friday, but I'm thoroughly confused.

Reply:
Apologies, I'd discussed the context for this change in the [slack thread](https://kubernetes.slack.com/archives/C0181L6JYQ2/p1685622093221839) but should have left a comment here. 

Essentially, with the change from `WithTimeout(5 * time.Minute)` to `WithTimeout(defaultTimeout)`, the tests were stepping on each other's toes. There was a ton of 
* "delete resources, and wait for upto 5 mins for the resources to be cleaned up"
* "do this, this and this to speed up the resource cleanup" 
etc etc. 

When I'd pushed this change, I'd thought about using a different catalog for this test, so the three specs in this suite can be independent from each other ([in line with ginkgo best practices](https://onsi.github.io/ginkgo/#:~:text=This%20allows%20Ginkgo%20to%20shuffle,and%20run%20specs%20in%20parallel.)). However, in the conversation in the slack thread, it became more apparent that not only will these tests benefit from using a different catalog, but they most likely belong to a different test suite altogether. Yet however, I noticed https://github.com/operator-framework/operator-controller/issues/247 which is converting these to unit tests anyway, so I left them commented out, so that https://github.com/operator-framework/operator-controller/issues/247 can just remove the commented out lines. 

Commenting them in in this PR requires either 
A) the building of another catalog with a new operator so that the specs can remain independent of each other, or 
B) having them run sequentially by adding safeguards such as "wait for resources to clean up" and "do a, b and c to speed up resource clean up" etc etc

option A will be wasted effort in this PR since 
a) a new bundle will be added when [plain bundle installation capability is added](https://github.com/operator-framework/operator-controller/pull/242)
b) the test is going away anyway. 

option B....doesn't help the "bring the overall time for the test suite run" objective of this PR at all. 

So leaving this commented out in this PR so that https://github.com/operator-framework/operator-controller/issues/247 can simply get rid of these commented out lines.